stellargraph/core/graph.py

```diff
@@ -279,6 +279,42 @@ def __init__(
             weight_column=edge_weight_column,
         )
 
+        if nodes is None:
+            nodes_from_edges = pd.unique(
```

Comment:
Hm, it turns out that we compute this value in either cases, and it's just a matter of when. I don't think it's worth optimising for the error case (e.g. if `convert_nodes` throws an exception, the `if nodes is not None` branch won't be taken), so we could potentially pull this out:

```python
# the actual nodes contained in the edges
nodes_from_edges = ...

if nodes is None:
    nodes_after_inference = pd.DataFrame([], index=nodes_from_edges)
else:
    nodes_after_inference = nodes

...

if nodes is not None:
    # ...
    try:
        self._nodes.ids.to_iloc(...)
    except KeyError as e:
        ...
```